test/types/helpers-isFilled.types.ts

```diff
@@ -0,0 +1,600 @@
+import { expectType, TypeEqual } from "ts-expect";
+
+import * as prismic from "../../src";
+
+/**
+ * RichText
+ */
+(value: prismic.RichTextField) => {
+	if (prismic.isFilled.richText(value)) {
+		expectType<TypeEqual<prismic.RichTextField<"filled">, typeof value>>(true);
+		expectType<TypeEqual<prismic.RichTextField<"empty">, typeof value>>(false);
+	} else {
+		expectType<TypeEqual<prismic.RichTextField<"filled">, typeof value>>(false);
+		expectType<TypeEqual<prismic.RichTextField<"empty">, typeof value>>(true);
+	}
+};
+
+/**
+ * Title
+ */
+(value: prismic.TitleField) => {
+	if (prismic.isFilled.title(value)) {
+		expectType<TypeEqual<prismic.TitleField<"filled">, typeof value>>(true);
+		expectType<TypeEqual<prismic.TitleField<"empty">, typeof value>>(false);
+	} else {
+		expectType<TypeEqual<prismic.TitleField<"filled">, typeof value>>(false);
+		expectType<TypeEqual<prismic.TitleField<"empty">, typeof value>>(true);
+	}
+};
+
+/**
+ * Image thumbnail
+ */
+(value: prismic.ImageFieldImage) => {
+	if (prismic.isFilled.imageThumbnail(value)) {
+		expectType<TypeEqual<prismic.ImageFieldImage<"filled">, typeof value>>(
+			true,
+		);
+		expectType<TypeEqual<prismic.ImageFieldImage<"empty">, typeof value>>(
+			false,
+		);
+	} else {
+		expectType<TypeEqual<prismic.ImageFieldImage<"filled">, typeof value>>(
+			false,
+		);
+		expectType<TypeEqual<prismic.ImageFieldImage<"empty">, typeof value>>(true);
+	}
+};
+
+/**
+ * Image
+ */
+
+// Default (without thumbnail)
+(value: prismic.ImageField) => {
+	if (prismic.isFilled.image(value)) {
+		expectType<TypeEqual<prismic.ImageField<never, "filled">, typeof value>>(
+			true,
+		);
+		expectType<TypeEqual<prismic.ImageField<never, "empty">, typeof value>>(
+			false,
+		);
+	} else {
+		expectType<TypeEqual<prismic.ImageField<never, "filled">, typeof value>>(
+			false,
+		);
+		expectType<TypeEqual<prismic.ImageField<never, "empty">, typeof value>>(
+			true,
+		);
+	}
+};
+
+// With thumbnails
+(value: prismic.ImageField<"foo">) => {
+	if (prismic.isFilled.image(value)) {
+		expectType<TypeEqual<prismic.ImageField<"foo", "filled">, typeof value>>(
+			true,
+		);
+		expectType<TypeEqual<prismic.ImageField<"foo", "empty">, typeof value>>(
+			false,
+		);
+	} else {
+		expectType<TypeEqual<prismic.ImageField<"foo", "filled">, typeof value>>(
+			false,
+		);
+		expectType<TypeEqual<prismic.ImageField<"foo", "empty">, typeof value>>(
+			true,
+		);
+	}
+};
```

Comment:
So there's an issue with the image field, again 😢

It fails to infer thumbnail names.

Reply:
I was able to fix this issue with a bit of a hack. 🥲🔨

As shared on Slack, `ImageField`'s State param in the thumbnails Record was falling back to FilledState ("filled" | "empty") rather than the actual param value. Using a ternary and a duplicated type seems to trigger a different method of evaluation that makes State the correct value.

See the comment here: https://github.com/prismicio/prismic-client/pull/265/commits/44369faf592e0dccffcc4ece93f5792f40217d2a